### PR TITLE
fix: add support for mssql_managed_instance

### PR DESCRIPTION
**Pull request by grixxie**
addressing issue: #2063

infracost currently supports the "deprecated" [sql_managed_instance](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/sql_managed_instance) -> add support for new [mssql_managed_instance](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mssql_managed_instance)

any feedback appreciated

### CODE REVIEW

**Comment by grixxie** [comment]
@tim775 I _think_ this is ready for review - I would appreciate some feedback / some guidance on what else is left to do here

File: internal/resources/azure/mssql_managed_instance.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+				AttributeFilters: ([]*schema.AttributeFilter{
+					{Key: "productName", Value: r.productDescription()},
+					{Key: "skuName", Value: r.meteredName()},
+				}),
+			},
+			PriceFilter: priceFilterConsumption,
+		},
+	}
+
+	costComponents = append(costComponents, r.mssqlMIStorageCostComponent(), r.mssqlMIBackupCostComponent())
```
Comment (tim775) [inline internal/resources/azure/mssql_managed_instance.go:80]: We should probably only add the storage cost component if it is above the default
```suggestion
	if r.StorageSizeInGb - 32 > 0 {
		costComponents = append(costComponents, r.mssqlMIStorageCostComponent(), r.mssqlMIBackupCostComponent())
	}
```

File: internal/resources/azure/mssql_managed_instance.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+			{Key: "long_term_retention_storage_gb", DefaultValue: 0, ValueType: schema.Int64},
+		},
+		CostComponents: costComponents,
+	}
+}
+
+func (r *MSSQLManagedInstance) costComponents() []*schema.CostComponent {
+	costComponents := []*schema.CostComponent{
+		{
+			Name:           fmt.Sprintf("Compute (%s %d Cores)", strings.ToTitle(r.SKU), r.Cores),
```
Comment (tim775) [inline internal/resources/azure/mssql_managed_instance.go:62]:
```suggestion
			Name:           fmt.Sprintf("Compute (%s %d cores)", strings.ToTitle(r.SKU), r.Cores),
```

**Review by tim775 — CHANGES_REQUESTED** [review]
Thanks @grixxie this is great!  
I left a few mostly style comments.  
Also we need to add the usage parameters to the [example usage file.](https://github.com/infracost/infracost/blob/master/infracost-usage-example.yml)

Let me know if you have time to make the changes, otherwise I'm happy to finish them off.
Cheers,
Tim

**Comment by grixxie** [comment]
Thanks for this @tim775 , I will make these changes including the example usage file :)

**Review by tim775 — APPROVED** [review]
Thanks, looks good!  If you would, please update the golden file test to pick up the changes by running:
```
ARGS="--run TestMSSQLManagedInstance -v -update" make test_azure
```

**Comment by grixxie** [comment]
Hey @tim775 , I'm still pretty new to this so I'd appreciate some help here - running ```make test_azure``` with the above commands is failing and the golden test file produced is showing all $0.00's.

I've followed the steps in CONTRIBUTING.md including the [azure-credentials](https://github.com/infracost/infracost/blob/master/CONTRIBUTING.md#azure-credentials) section but still no luck. I'm unsure whether it's a problem with my code changes or my local setup at this point - maybe you can help?

Apologies, still getting into this :laughing:

**Comment by tim775** [comment]
@grixxie, no worries I actually ended up merging this with the intention of updating the golde file test after and ran into the same issue you're seeing now.  It looks like the filters got messed up somehow, but it was a [quick fix.](https://github.com/infracost/infracost/pull/2422/files#diff-e13abd66b50877fed0bdd48c862d36d8564d12f5947f38021034d06363d4539f)

**Pull request merged**